af/fastpath/fastpath/db.py

```diff
@@ -40,9 +40,12 @@ def _ping():
         log.info("Database start time: %s", row[0])
 
 
-def setup():
+def setup(conf) -> None:
     global conn, _autocommit_conn
-    dsn = f"host={DB_HOST} user={DB_USER} dbname={DB_NAME} password={DB_PASSWORD}"
+    if conf.db_uri:
+        dsn = conf.db_uri
+    else:
+        dsn = f"host={DB_HOST} user={DB_USER} dbname={DB_NAME} password={DB_PASSWORD}"
```

Comment:
Maybe this can be refactored as:
`<bikeshed>`
```
    dsn = f"host={DB_HOST} user={DB_USER} dbname={DB_NAME} password={DB_PASSWORD}"
    if conf.db_uri:
        dsn = conf.db_uri
```